### PR TITLE
Refactor make commands and workflows on CI

### PR DESCRIPTION
**Pull request by nghialv**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: .github/workflows/publish.yaml [hunk 1]
```diff
@@ -33,7 +35,7 @@ jobs:
       # Building frontend.
```
Comment (knanao) [inline .github/workflows/publish.yaml:35]:
```suggestion
      # Building web.
```

File: .github/workflows/publish.yaml [hunk 2]
```diff
@@ -47,8 +49,7 @@ jobs:
       - name: Update frontend dependencies
         run: make update/frontend-deps
       - name: Build web static
-        run: |
-          make build/frontend
+        run: make build/frontend
```
Comment (knanao) [inline .github/workflows/publish.yaml:52]:
```suggestion
        run: make build/web
```

File: .github/workflows/publish.yaml [hunk 3]
```diff
@@ -47,8 +49,7 @@ jobs:
       - name: Update frontend dependencies
         run: make update/frontend-deps
```
Comment (knanao) [inline .github/workflows/publish.yaml:50]:
```suggestion
      - name: Update web dependencies
        run: make update/web-deps
```

File: .github/workflows/publish.yaml (hunk 1 above)
Comment (nghialv) [reply .github/workflows/publish.yaml:35]: Nice catch!

File: .github/workflows/publish.yaml (hunk 3 above)
Comment (nghialv) [reply .github/workflows/publish.yaml]: @knanao Addressed it. https://github.com/pipe-cd/pipecd/pull/3585/commits/d4611ce58b1eaf46bc22f312762d549a9e7fbcfd

File: .github/workflows/publish.yaml [hunk 4]
```diff
@@ -22,18 +24,18 @@ jobs:
           fetch-depth: 0
       - uses: actions/setup-go@v2
         with:
-          go-version: '1.17.6'
+          go-version: ${{ env.GO_VERSION }}
       - name: Determine version
         run: echo "PIPECD_VERSION=$(git describe --tags --always --abbrev=7)" >> $GITHUB_ENV
 
       # Building backend.
```
Comment (knanao) [inline .github/workflows/publish.yaml:31]: Sorry for the many times, I'll leave some nits.
it's up to you, whether choosing `go` or `Go`. lol
```suggestion
      # Building go.
```

**Review by khanhtc1202 — APPROVED** [review]
🙌

File: .github/workflows/publish.yaml [hunk 5]
```diff
@@ -22,18 +24,18 @@ jobs:
...
       - name: Build backend modules
         run: make build/backend
```
Comment (knanao) [inline .github/workflows/publish.yaml:33]:
```suggestion
      - name: Build go modules
        run: make build/go
```

**Review by knanao — APPROVED** [review]

**Pull request merged**